### PR TITLE
Sha1 attributes

### LINKED ISSUE #125: Generalization of sha1.default
As I was trying to improve the `sha1` for the PKNCA package, I found that I needed `sha1.sessionInfo()` and `sha1.packageDescription()` functions.  Those would be just be:

```r
sha1.sessionInfo <- digest:::sha1.list
sha1.packageDescription <- digest:::sha1.list
```

That made me think that a generalization of the `sha1.default()` function to check with something like the following would help:

```r
if (is.list(x)) {
  sha1.list(x=x, digits = digits, zapsmall = zapsmall, ..., algo = algo)
} else {
  # The current stop message
}
```

And, the one other change that may be of interest would be that `sha1.list()` pull out the attributes of the list for checking, too.  (Though, that looks like a different issue that I'll open separately.)

https://github.com/eddelbuettel/digest/blob/4c415fe2b67b760f7cd64767620d561a5a1a0ed6/R/sha1.R#L28-L35

Would such a PR be of interest, or would you prefer the specific methods for each class?  (And therefore you would prefer a PR that explicitly names the sessionInfo and packageDescription classes.)

Tag: @ThierryO, @eddelbuettel

### PR DESCRIPTION
**Pull request by ThierryO**
fixes #125 and #126

### CODE REVIEW

**Comment by eddelbuettel** [comment]
Looks very clean, thank you!   (I may tweak the docs and version comparison to make 0.6.23 as interim versions "do not really exist" in the on-CRAN and in  the big wide world sense.)

**Comment by eddelbuettel** [comment]
The codecov change is from the is.list check:  

```r
    if (is.list(x)) {
        return(
            sha1.list(x, digits = digits, zapsmall = zapsmall, ..., algo = algo)
        )
    }
```

Can you throw a quick unit test in for a list object ?

**Comment by eddelbuettel** [comment]
Done now in https://github.com/eddelbuettel/digest/pull/127/commits/59ba9fb4268411866759bec71b60faa24d0ebdd0

**Comment by eddelbuettel** [comment]
The test not working reveals that there is of course a `sha1.list()` method so will that code in `sha1.default()` ever work?

**Comment by ThierryO** [comment]
a `sessionInfo()` is a list like object and shoudl be handled by `sha1.default()`

File: inst/tinytest/test_sha1.R [hunk 1]
```diff
@@ -467,3 +468,19 @@ for (algo in c("md5", "sha1", "crc32", "sha256", "sha512", "xxhash32",
     )
 
 }
+
```
Comment (ThierryO) [inline inst/tinytest/test_sha1.R]: unit test for `sha.default()` with `is.list(x) == TRUE`
```suggestion
expect_true(is.character(sha1(sessionInfo())))
```

**Comment by ThierryO** [comment]
I've added the unit test with an interactive rebase to fe2f6f8b.
The documentation update (65d1875d) now states versions 0.6.22 and 0.6.23

**Comment by eddelbuettel** [comment]
I still see the 'four-part' version in the rd file but no worries, I can smooth over that when I roll up the next release (unless I forget).

Thanks again--will merge.

**Comment by eddelbuettel** [comment]
> I still see the 'four-part' version in the rd file 

Must have been a web-page caching artefact. All good.  Thanks again!

**Comment by billdenney** [comment]
Thanks to both of you for your help on these issues!

**Pull request merged**